src/Workspaces/Remote/ServiceHub/Host/AssetProvider.cs

```diff
@@ -205,57 +205,62 @@ static void AddAll(HashSet<Checksum> checksums, ChecksumCollection checksumColle
             var usePool = missingChecksumsCount <= PooledChecksumArraySize;
             var missingChecksums = usePool ? s_checksumPool.Allocate() : new Checksum[missingChecksumsCount];
 
-            missingChecksumsCount = 0;
-            foreach (var checksum in checksums)
+            try
```

Comment:
view with whitespace off.

Reply:
💭 Seems like the pool could just be removed in favor of `stackalloc` for the initial array.

Reply:
Ah, nevermind due to callbacks

Reply:
and async/await :(